src/amo/components/UserProfileEdit/index.js

```diff
@@ -0,0 +1,364 @@
+      return;
+    }
+
+    dispatch(editUserAccount({
+      errorHandlerId: errorHandler.id,
+      userFields: {
+        biography: getValueOrEmpty(this.biography),
+        display_name: getValueOrEmpty(this.displayName),
+        homepage: getValueOrEmpty(this.homepage),
+        location: getValueOrEmpty(this.location),
+        occupation: getValueOrEmpty(this.occupation),
+        username: getValueOrEmpty(this.username),
+      },
+      userId: user.id,
+    }));
+  }
+
+  biography: HTMLInputElement | null;
+  homepage: HTMLInputElement | null;
+  location: HTMLInputElement | null;
+  occupation: HTMLInputElement | null;
+  username: HTMLInputElement | null;
+
+  render() {
+    const {
+      currentUser,
+      errorHandler,
+      hasEditPermission,
+      i18n,
+      isEditing,
+      isEditingCurrentUser,
+      user,
+      username,
+    } = this.props;
+
+    if (currentUser && user && !hasEditPermission) {
+      return <NotFound />;
+    }
+
+    return (
+      <div className="UserProfileEdit">
+        {user && (
+          <Helmet>
+            <title>
+              {i18n.sprintf(
+                i18n.gettext('User Profile for %(user)s'),
+                { user: user.displayName }
+              )}
+            </title>
+          </Helmet>
+        )}
+
+        <Card className="UserProfileEdit-user-links">
+          <ul>
+            <li>
+              <Link to={`/user/${username}/`}>
+                {isEditingCurrentUser ?
+                  i18n.gettext('View My Profile') :
+                  i18n.gettext("View user's profile")
+                }
+              </Link>
+            </li>
+            <li>
+              {isEditingCurrentUser ?
+                i18n.gettext('Edit My Profile') :
+                i18n.gettext("Edit user's profile")
+              }
+            </li>
+          </ul>
+        </Card>
+
+        <form
+          action=""
+          className="UserProfileEdit-form"
+          onSubmit={this.onSubmit}
+        >
+          <div>
+            {errorHandler.renderErrorIfPresent()}
+
+            <Card
+              className="UserProfileEdit--Card"
+              header={isEditingCurrentUser ? i18n.gettext('Your Account') : (
+                i18n.sprintf(i18n.gettext('User Account for %(username)s'), {
+                  username,
+                })
+              )}
+            >
+              <p className="UserProfileEdit-aside">
+                {i18n.gettext(`Manage basic account information, such as your
+                  username and Firefox Accounts settings.`)}
+              </p>
+
+              <label className="UserProfileEdit--label" htmlFor="username">
+                {i18n.gettext('Username')}
+              </label>
+              <input
+                className="UserProfileEdit-username"
+                disabled={!user}
+                id="username"
+                name="username"
+                ref={(ref) => { this.username = ref; }}
```

Comment:
This is more of a question than a comment. I see the way we're managing this form's state is different from what we've done with the collection edit form. We're using `ref` and `defaultValue`, whereas in the [collection form](https://github.com/mozilla/addons-frontend/blob/72fb82c57cb50815caf16719d30a0d585538cb74/src/amo/components/CollectionManager/index.js#L145-L151) we use an internal state and an `onChange` function to manage it.

What are the pros and cons between the two approaches?

Reply:
How consistent is the "uncontrolled" approach with the other forms we already have? I'd prefer  one way of doing things over two for consistency across the codebase.